main/persistent_storage/persistent_storage.h

```diff
@@ -40,22 +42,13 @@ esp_err_t persistent_storage_set_str(
 	size_t max_string_length
 );
 
-/**
- * Store the WiFi credentials in Persistent Storage
- *
- * @param ssid The WiFi SSID
- * @param password The WiFi password
- */
-esp_err_t persistent_storage_set_wifi(const char* ssid, const char* password);
+esp_err_t persistent_storage_set_connection_data(
+	connection_data_t connection_data
+);
 
 /**
- * Get WiFi credentials from Persistent Storage
- *
- * @param ssid A character buffer in which the WiFi SSID will be stored. It
- * needs to be at least WIFI_SSID_CHAR_BUFFER_LENGTH long
- * @param password A character buffer in which the WiFi Password will be stored.
- * It needs to be at least WIFI_PASSWORD_CHAR_BUFFER_LENGTH long
+ * Erase the default NVS Flash partition
  */
-esp_err_t persistent_storage_get_wifi(char* ssid, char* password);
```

Comment:
How will I retrieve the data now? There's no replacement function for  this, could you write a persistent_storage_get_connection_data